File: surf-api-bukkit/surf-api-bukkit-server/src/main/java/dev/slne/surf/surfapi/bukkit/server/impl/nms/listener/packets/serverbound/RenameItemPacketImpl.java
package dev.slne.surf.surfapi.bukkit.server.impl.nms.listener.packets.serverbound;

import dev.slne.surf.surfapi.bukkit.api.nms.listener.packets.serverbound.RenameItemPacket;
import net.minecraft.network.protocol.game.ServerboundRenameItemPacket;

public final class RenameItemPacketImpl extends NmsServerboundPacketImpl<ServerboundRenameItemPacket> implements
    RenameItemPacket {

  public RenameItemPacketImpl(ServerboundRenameItemPacket nmsPacket) {
    super(nmsPacket);
  }

  @Override
  public String getNewName() {
    return getNmsPacket().getName();
  }
}

File: surf-api-bukkit/surf-api-bukkit-server/src/main/kotlin/dev/slne/surf/surfapi/bukkit/server/impl/nms/listener/packets/serverbound/RenameItemPacketImpl.kt
package dev.slne.surf.surfapi.bukkit.server.impl.nms.listener.packets.serverbound

import dev.slne.surf.surfapi.bukkit.api.nms.NmsUseWithCaution
import dev.slne.surf.surfapi.bukkit.api.nms.listener.packets.serverbound.RenameItemPacket
import net.minecraft.network.protocol.game.ServerboundRenameItemPacket

@NmsUseWithCaution
class RenameItemPacketImpl(nmsPacket: ServerboundRenameItemPacket) :
    NmsServerboundPacketImpl<ServerboundRenameItemPacket>(nmsPacket), RenameItemPacket {
    override val newName: String get() = nmsPacket.name
}
